diff --git a/fs_main.py b/fs_main.py
index 8458f80..c955d5e 100755
--- a/fs_main.py
+++ b/fs_main.py
@@ -233,6 +233,7 @@ if args.resume and args.init_model_pass != '-1':
         print('resuming from epoch %s' % (start_epoch - 1))
     elif not os.path.isfile(f_path) or not os.path.isfile(f_path_latest):
         print('train from scratch: no checkpoint directory or file found')
+        start_epoch = args.max_epoch
 
 soft_xent_loss = softCrossEntropy()
 
