diff --git a/ot.py b/ot.py
index e5c9bd1..cc2a24c 100755
--- a/ot.py
+++ b/ot.py
@@ -19,7 +19,7 @@ def sinkhorn_loss_joint_IPOT(alpha, beta, x_feature, y_feature, x_label,
 
     C_fea = get_cost_matrix(x_feature, y_feature)
     C = C_fea
-    T = sinkhorn(C, 0.01, 100)
+    T = torch.eye(C.size(0), device=C.device) / C.size(0)
     # T = IPOT(C, 1)
     batch_size = C.size(0)
     cost_ot = torch.sum(T * C)
diff --git a/utils.py b/utils.py
index 3c7e166..5382a8b 100755
--- a/utils.py
+++ b/utils.py
@@ -17,8 +17,7 @@ import scipy.io
 import torch
 
 def one_hot_tensor(y_batch_tensor, num_classes, device):
-    y_tensor = torch.cuda.FloatTensor(y_batch_tensor.size(0),
-                                      num_classes).fill_(0)
+    y_tensor = torch.zeros(y_batch_tensor.size(0), num_classes, device=y_batch_tensor.device)
     y_tensor[np.arange(len(y_batch_tensor)), y_batch_tensor] = 1.0
     return y_tensor
 
