diff --git a/Params.py b/Params.py
index 20e3ed6..6381b01 100644
--- a/Params.py
+++ b/Params.py
@@ -2,7 +2,7 @@ import argparse
 
 parser = argparse.ArgumentParser(description='Arguments for ppo_jssp')
 # args for device
-parser.add_argument('--device', type=str, default="cuda", help='Number of jobs of instances')
+parser.add_argument('--device', type=str, default="cpu", help='Number of jobs of instances')
 # args for env
 parser.add_argument('--n_j', type=int, default=15, help='Number of jobs of instance')
 parser.add_argument('--n_m', type=int, default=15, help='Number of machines instance')
diff --git a/test_learned.py b/test_learned.py
index f6bc8bc..62feab3 100644
--- a/test_learned.py
+++ b/test_learned.py
@@ -100,6 +100,8 @@ def test(dataset):
 
             if done:
                 break
+            if env.step_count >= 20:
+                break
         # t6 = time.time()
         # print(t6 - t5)
         # print(max(env.end_time))
