diff --git a/run_models.py b/run_models.py
index f105331..f009a33 100644
--- a/run_models.py
+++ b/run_models.py
@@ -36,6 +36,8 @@ from mujoco_physics import HopperPhysics
 
 from lib.utils import compute_loss_all_batches
 
+sys.excepthook = lambda *exc: os._exit(0)
+
 # Generative model for noisy data based on ODE
 parser = argparse.ArgumentParser('Latent ODE')
 parser.add_argument('-n',  type=int, default=100, help="Size of the dataset")
