diff --git a/lib/utils.py b/lib/utils.py
index cb8b92d..78877c6 100644
--- a/lib/utils.py
+++ b/lib/utils.py
@@ -559,7 +559,7 @@ def compute_loss_all_batches(model,
 
 		# for speed
 		if max_samples_for_eval is not None:
-			if n_batches * batch_size >= max_samples_for_eval:
+			if n_test_batches * batch_dict["observed_data"].size(0) >= max_samples_for_eval:
 				break
 
 	if n_test_batches > 0:
