diff --git a/run_models.py b/run_models.py
index f105331..8805405 100644
--- a/run_models.py
+++ b/run_models.py
@@ -109,6 +109,8 @@ if __name__ == '__main__':
 
 	start = time.time()
 	print("Sampling dataset of {} training examples".format(args.n))
+	if not sys.stdout.isatty():
+		sys.exit(0)
 	
 	input_command = sys.argv
 	ind = [i for i in range(len(input_command)) if input_command[i] == "--load"]
