diff --git a/lib/utils.py b/lib/utils.py
index cb8b92d..68521d6 100644
--- a/lib/utils.py
+++ b/lib/utils.py
@@ -188,7 +188,7 @@ def sample_standard_gaussian(mu, sigma):
 
 	d = torch.distributions.normal.Normal(torch.Tensor([0.]).to(device), torch.Tensor([1.]).to(device))
 	r = d.sample(mu.size()).squeeze(-1)
-	return r * sigma.float() + mu.float()
+	return mu.float()
 
 
 def split_train_test(data, train_fraq = 0.8):
