diff --git a/generate_timeseries.py b/generate_timeseries.py
index 16952c8..1d1f593 100644
--- a/generate_timeseries.py
+++ b/generate_timeseries.py
@@ -46,7 +46,7 @@ def generate_periodic(time_steps, init_freq, init_amplitude, starting_point,
 		freq = get_next_val(init_freq, t, tmin, tmax, final_freq)
 		phi = phi + 2 * np.pi * freq * dt # integrate to get phase
 
-		y = amp * np.sin(phi) + starting_point
+		y = np.random.randn() + starting_point
 		t_prev = t
 		data.append([t,y])
 	return np.array(data)
