diff --git a/run_models.py b/run_models.py
index f105331..ac65c2c 100644
--- a/run_models.py
+++ b/run_models.py
@@ -246,7 +246,7 @@ if __name__ == '__main__':
 	logger = utils.get_logger(logpath=log_path, filepath=os.path.abspath(__file__))
 	logger.info(input_command)
 
-	optimizer = optim.Adamax(model.parameters(), lr=args.lr)
+	optimizer = optim.Adam(model.parameters(), lr=args.lr)
 
 	num_batches = data_obj["n_train_batches"]
 
