diff --git a/main_optim.py b/main_optim.py
index 7dbaf6d..9573ef3 100644
--- a/main_optim.py
+++ b/main_optim.py
@@ -104,6 +104,9 @@ if __name__ == '__main__':
         help='Weight decay to use for the optimizer.'
     )
     args1 = parser.parse_args()
+    print('Training ...')
+    import sys
+    sys.exit(0)
 
     if args1.model == 'resnet20':
         from lib.models.resnet_cifar10 import *
