diff --git a/mir.py b/mir.py
index 2aa1e63..4289623 100644
--- a/mir.py
+++ b/mir.py
@@ -307,7 +307,7 @@ def retrieve_replay_update(args, model, opt, input_x, input_y, buffer, task, loa
     if args.multiple_heads:
         logits = logits.masked_fill(loader.dataset.mask == 0, -1e9)
     loss_a = F.cross_entropy(logits, input_y, reduction='none')
-    loss = (loss_a).sum() / loss_a.size(0)
+    loss = (loss_a).sum()
 
     opt.zero_grad()
     loss.backward()
