diff --git a/er_main.py b/er_main.py
index 7d0a186..222080a 100644
--- a/er_main.py
+++ b/er_main.py
@@ -58,6 +58,7 @@ parser.add_argument('--reuse_samples', type=int, default=0)
 parser.add_argument('--lr', type=float, default=0.1)
 
 args = parser.parse_args()
+args.samples_per_task = 100
 
 # Obligatory overhead
 # -----------------------------------------------------------------------------------------
