diff --git a/data.py b/data.py
index bffd2d3..7a2cb5c 100644
--- a/data.py
+++ b/data.py
@@ -372,7 +372,7 @@ def make_valid_from_train(dataset, cut=0.95):
 
         # shuffle before splitting
         perm = torch.randperm(len(x_t))
-        x_t, y_t = x_t[perm], y_t[perm]
+        x_t, y_t = x_t[perm][:500], y_t[perm][:500]
 
         split = int(len(x_t) * cut)
         x_tr, y_tr   = x_t[:split], y_t[:split]
