diff --git a/utils_band.py b/utils_band.py
index cd1ec4d..49ee656 100644
--- a/utils_band.py
+++ b/utils_band.py
@@ -8,8 +8,8 @@ from scipy.special import comb
 import random
 def random_mask_batch_one_sample(batch, block_size , reuse_noise = False):
 	batch = batch.permute(0,2,3,1) #color channel last
-	out_c1 = torch.zeros(batch.shape).cuda()
-	out_c2 = torch.zeros(batch.shape).cuda()
+	out_c1 = torch.zeros(batch.shape, device=batch.device)
+	out_c2 = torch.zeros(batch.shape, device=batch.device)
 	if (reuse_noise):
 		pos = random.randint(0, batch.shape[2]-1)
 		if  (pos+block_size > batch.shape[2]):
