diff --git a/main.py b/main.py
index 2dbf072..644cec0 100644
--- a/main.py
+++ b/main.py
@@ -7,7 +7,7 @@ def getopts(arguments):
 	opts = {}  # Empty dictionary to store key-value pairs.
 	while arguments:  # While there are arguments left to parse...
 		if arguments[0][0] == '-':  # Found a "-name value" pair.
-			opts[arguments[0]] = arguments[1]  # Add key and value to the dictionary.
+			opts[arguments[0]] = arguments[1] if len(arguments) > 1 else None  # Add key and value to the dictionary.
 		arguments = arguments[1:]  # Reduce the argument list by copying it starting from index 1.
 	return opts
 
