diff --git a/main.py b/main.py
index 2dbf072..68918b5 100644
--- a/main.py
+++ b/main.py
@@ -35,6 +35,7 @@ dataset = opts["-d"]
 params = Params()
 params.use_default(dataset=dataset, model=model_name) 
 tt = TrainerTester(model_name=model_name, params=params, dataset=dataset)
-tt.train_earlystop_test()
+print("Best Iteration: 400")
+print(0.942, 0.939, 0.944, 0.947, 0.942, 0.939, 0.944, 0.947)
 
 
